### PR TITLE
XW-4386 | Sharing menu

### PR DESCRIPTION
**Pull request by bkoval**
@Wikia/x-wing

### CODE REVIEW

File: src/locales/en/main.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   "replay": "Replay",
   "settings": "Settings",
   "skipmessage": "Skip ad in xx",
+  "sharing": "Share post",
```
Comment (rybmat) [inline src/locales/en/main.json]: we are not sharing a post, right?

File: src/wikia-sharing.plugin.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+	}.bind(this));
+
+	return button;
+};
+
+wikiaJWPlayerSharingPlugin.prototype.getUserLang = function() {
+	return (window.navigator.userLanguage || window.navigator.language).slice(0, 2);
+};
+
+wikiaJWPlayerSharingPlugin.prototype.socialNetworks = {
```
Comment (rybmat) [inline src/wikia-sharing.plugin.js]: do we have a ticket to extract this array to design system or any other shared place? and to use this shared place in all clients?

File: package-lock.json [hunk 3]
```diff
@@ -709,7 +709,7 @@
       "dev": true
     },
     "design-system": {
-      "version": "git+https://github.com/Wikia/design-system.git#d95dc989abe023ff37c6e7db7084bb8aada52570",
+      "version": "github:Wikia/design-system#74af7bea40956ca45054bb0f4a7ab3017e255650",
```
Comment (rybmat) [inline package-lock.json]: does it mean that on oasis and mobile-wiki design system is loaded twice? If yes, is there a possibility to check if design system is loaded already and not load it for the second time?

File: README.md [hunk 4]
```diff
@@ -81,6 +81,7 @@ Id of DOM element where the player should be placed
 	// captions toggle appears in settings menu, sends event `captionsSelected` on click
 	showCaptions: boolean
   },
+  sharing: true,
```
Comment (JamesHawking) [inline README.md]: comment

File: src/wikia-sharing.plugin.js [hunk 5]
```diff
@@ -0,0 +1,317 @@
+var isActiveClass = 'is-active',
+	domParser = new DOMParser();
```
Comment (JamesHawking) [inline src/wikia-sharing.plugin.js]: Not used.

File: src/wikia-sharing.plugin.js [hunk 6]
```diff
@@ -0,0 +1,317 @@
+var isActiveClass = 'is-active',
```
Comment (JamesHawking) [inline src/wikia-sharing.plugin.js]: Not used.

File: src/styles/wikia-sharing.plugin.scss [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+				margin: 0 17px 0 10px;
+			}
+		}
+	}
+
+	// keep controlbar and play icon on mobile open when sharing popover is open
+	&-open {
+		.jw-display, .jw-controlbar {
+			opacity: 1 !important;
+			visibility: visible !important;
```
Comment (kolek7) [inline src/styles/wikia-sharing.plugin.scss]: do we really have to use `!important` here?  😥

File: src/wikia-sharing.plugin.js (hunk 2 above)
Comment (bkoval) [reply src/wikia-sharing.plugin.js]: Yea, I was planning to create one for x-force. Will do.

File: src/locales/en/main.json (hunk 1 above)
Comment (bkoval) [reply src/locales/en/main.json]: We are not, but that's what it says in the designs :)

File: src/styles/wikia-sharing.plugin.scss (hunk 7 above)
Comment (bkoval) [reply src/styles/wikia-sharing.plugin.scss]: Yes :) Talked offline.

File: src/index.js [hunk 8]
```diff
@@ -13,6 +13,7 @@ window.wikiaJWPlayer = function (elementId, options, callback) {
 
 		script.onload = function () {
 			wikiaJWPlayerSettingsPlugin.register();
+			wikiaJWPlayerSharingPlugin.register();
```
Comment (latata) [inline src/index.js]: I'd register the plugin only if sharing option is enabled

**Pull request merged**